core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -98,8 +98,11 @@ public void commit(TableMetadata base, TableMetadata metadata) {
       LOG.info("Nothing to commit.");
       return;
     }
+    TableMetadata updated = (base != null && base.file() != null) ?
+            metadata.addPreviousMetadata(base.file().location(), base.lastUpdatedMillis()) : metadata;
```

Comment:
This kind of concern should be handled in the `TableMetadata` update methods. This is similar to `metadata.rollbackTo`, which not only sets the current version, but also updates the table history. I think that all of those methods should add the current object's `file.location()` to the previous metadata location list, if it is non-null. That way, the caller never needs to remember to update it.

Reply:
@rdblue Initially thought of handling in the `TableMetadata`, but I went with `TableOperations` because it just two places to track previous versions. Otherwise every method and new method in future should add current file location.

But I agree, should be handled in the `TableMetadata`, will change it.